File: application/src/main/java/com/example/server/repository/entity/Comment.java
package com.example.server.repository.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@AllArgsConstructor
@Entity
@EqualsAndHashCode(of = "id")
@Getter
@NoArgsConstructor
@Setter
@SequenceGenerator(name = "comments_id_seq", allocationSize = 1)
@Table(name = "comments")
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "comments_id_seq")
    private Long id;

    @Column(name = "text")
    private String text;

    @JoinColumn(name = "item_id")
    @ManyToOne
    private Item item;

    @JoinColumn(name = "author_id")
    @ManyToOne
    private User author;

    @Column(name = "created")
    @CreationTimestamp
    private LocalDateTime created;

    public boolean hasSameItem(Item item) {
        return getItem().getId().equals(item.getId());
    }
}

File: application/src/main/kotlin/com/example/server/repository/entity/Comment.kt
package com.example.server.repository.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@SequenceGenerator(name = "comment_id_seq", allocationSize = 1)
@Table(name = "comment")
class Comment(

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "comment_id_seq")
    var id: Long?,

    @Column(name = "text")
    var text: String,

    @JoinColumn(name = "item_id")
    @ManyToOne
    var item: Item,

    @JoinColumn(name = "author_id")
    @ManyToOne
    var author: User,

    @Column(name = "created")
    @CreationTimestamp
    var created: LocalDateTime?,
)
